backend/database-fs.js

```diff
@@ -1,17 +1,16 @@
 'use strict'
 
-import { mkdir, readdir, readFile, unlink, writeFile } from 'node:fs/promises'
-import { join, resolve } from 'node:path'
+import { mkdir, readdir, readFile, rm, unlink, writeFile } from 'node:fs/promises'
+import { dirname, join, resolve } from 'node:path'
 import { checkKey } from '~/shared/domains/chelonia/db.js'
-
-// Initialized in `initStorage()`.
-let dataFolder = ''
+import DatabaseBackend from './DatabaseBackend.js'
 
 // Some operating systems (such as macOS and Windows) use case-insensitive
 // filesystems by default. This can be problematic for Chelonia / Group Income,
 // as we rely on keys being case-sensitive. This is especially relevant for CIDs,
 // where collisions could lead to DoS or data corruption.
-async function testCaseSensitivity () {
+async function testCaseSensitivity (backend: Object) {
```

Comment:
Going with the OOP approach, this makes more sense as a class method, private if you'd like.

Reply:
I didn't make it a class method to minimize diff changes. Another reason is I think that function could turn out to be useful somewhere else.

Reply:
> Another reason is I think that function could turn out to be useful somewhere else.

Sure, but in that case the `backend` argument seems odd. Since it's only used internally, I think it's fine as is though.